### PR TITLE
Added Lifestreet adapter.

### PR DESCRIPTION
**Pull request by Lifestreet**
## Type of change

- [x] New bidder adapter

## Description of change

- test parameters for validating bids
```
{
  bidder: 'lifestreet',
  params: {
    jstag_url: '//ads.lfstmedia.com/getad?site=285071',
    slot: 'slot166704',
    adkey: '78c',
    ad_size: '160x600',
    timeout: 700
  }
}
```
- contact email of the adapter’s maintainer
mobile.development@lifestreet.com
- [x] official adapter submission

### CODE REVIEW

File: src/adapters/lifestreet.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        LSM_Slot(slotTagParams);
+      } else {
+        _addSlotBidResponse(placementCode, 0, null, 0, 0);
+      }
+    });
+  }
+
+  function _addSlotBidResponse(placementCode, cpm, ad, width, height) {
+    let hasResponse = cpm && ad && ad.length > 0;
+    let bidObject = bidfactory.createBid(hasResponse ? 1 : 2);
```
Comment (jaiminpanchal27) [inline src/adapters/lifestreet.js]: Add bid request as second param to match request response pair.

File: src/adapters/lifestreet.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          _hb_request: ADAPTER_VERSION,
+          _timeout: timeout,
+          _onload: (slot, action, cpm, width, height) => {
+            if (slot.state() !== 'error') {
+              let slotName = slot.getSlotObjectName();
+              $$PREBID_GLOBAL$$[slotName] = slot;
+              if (slotName && !SLOTS_LOAD_MAP[slotName]) {
+                SLOTS_LOAD_MAP[slotName] = true;
+                let ad = `<div id="LSM_AD"></div>
+                    <script>window.parent.$$PREBID_GLOBAL$$["` + slotName + `"]
```
Comment (jaiminpanchal27) [inline src/adapters/lifestreet.js]: window.parent will not work in safeframe.

**Comment by Lifestreet** [comment]
Regarding SafeFrame: AFAIK these are not supported in Prebid.js currently. If it's an important issue, could you please suggest what is the best way to access $$PREBID_GLOBAL$$ from the creative when it's needed?

**Comment by jaiminpanchal27** [comment]
@protonate can you answer this one ?

**Comment by protonate** [comment]
SafeFrame support was implemented here: 3ec356fadbfb4d8f500999211537ac44df96aa03

This is included in Prebid 0.18.0.

Have a look at #955 for an improved example, this PR will merge this release.

**Comment by Lifestreet** [comment]
Thanks @protonate and @jaiminpanchal27

**Comment by jaiminpanchal27** [comment]
Looks good. Thanks for the changes @Lifestreet

**Comment by jaiminpanchal27** [comment]
This is merged into master. Please submit a PR to the [docs repo](https://github.com/prebid/prebid.github.io) to add a file for your adapter to the [bidders directory](https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders) so your adapter's params will appear on the [bidders page](http://prebid.org/dev-docs/bidders.html). Thank you for contributing

**Comment by Lifestreet** [comment]
Thank you. Will do! Could you please let us know if there is anything additional required to be approved for the list of header bidding partners on the download page (http://prebid.org/download.html)?

**Comment by aneuway2** [comment]
Hi @Lifestreet !
Are all of your parameters required to use the Life Street adaptor?

Thanks!

**Comment by Lifestreet** [comment]
Hi @aneuway2 !
A timeout **parameter** is optional, but preferable. The rest of parameters are required.

Thank you.

**Pull request merged**